components/tidb_query_vec_expr/src/impl_compare.rs

```diff
@@ -268,7 +268,7 @@ pub fn greatest_time(ctx: &mut EvalContext, args: &[&Option<Bytes>]) -> Result<O
                     Err(_) => {
                         return ctx
                             .handle_invalid_time_error(Error::invalid_time_format(&s))
-                            .map(|_| Ok(None))?;
+                            .map(|_| Ok(Some(s.to_owned().into_bytes())))?;
```

Comment:
@sticnarf If I make this change to LeastTime function, it will pass the push-down test which reported output wrong before. I also checked the TiDB version [LeastTime function](https://github.com/pingcap/tidb/blob/master/expression/builtin_compare.go#L813). Is this a bug or something? Shouldn't it be None/NULL if you use LeastTime to compare an invalid datetime string?

Reply:
The TiDB code you post does return `isNull = true` and an error. I think the problem is not here...

Reply:
I think he problem is that you return `None` as soon as any one of the input is invalid. But actually, the invalid ones can be ignored.